Source/Fuse.Controls.Native/iOS/ImageView.uno

```diff
@@ -13,22 +13,44 @@ namespace Fuse.Controls.Native.iOS
 
 	extern(iOS) internal class ImageView : LeafView, IImageView
 	{
+
+		ImageSource _imageSource;
 		public ImageSource ImageSource
 		{
 			set
 			{
 				if (value == null)
 					ImageHandle = null;
-				else if (value is FileImageSource)
-					UpdateImage((FileImageSource)value);
-				else if (value is HttpImageSource)
-					UpdateImage((HttpImageSource)value);
-				else if (value is MultiDensityImageSource)
-					UpdateImage((MultiDensityImageSource)value);
-				else
-				{
-					throw new Exception(value + " not supported in native context");
-				}
+				_imageSource = value;
+				OnImageSourceChanged();
+			}
+			private get
+			{
+				return _imageSource;
+			}
+		}
+
+		void OnImageSourceChanged()
+		{
+			if (ImageSource == null)
+				return;
+			else if (ImageSource is FileImageSource)
+				UpdateImage((FileImageSource)ImageSource);
+			else if (ImageSource is HttpImageSource)
+				UpdateImage((HttpImageSource)ImageSource);
+			else if (ImageSource is MultiDensityImageSource)
+			{
+				if (ImageSource != null && ImageSource is MultiDensityImageSource)
```

Comment:
This cleanup won't happen if the user switches from a MultiDensityImageSource to a different type of ImageSource.
A common pattern in fuselibs is on settting a new value, cleanup the old one first, regardless of what it is, then handle the new value.  You've kind of mixed the two steps together here.